### PR TITLE
[fixed] hidden check was breaking tabbing through forms

### LINKED ISSUE #211: I can't navigate trough form elements using tab button
### Steps to reproduce:
1. add some form elements inside a popup
2. try to navigate with tab button
### Expected behavior:

focus should move trought the form elements

### PR DESCRIPTION
**Pull request by evanhobbs**
Fixes #211 .

This is regarding: https://github.com/reactjs/react-modal/issues/211 and credit goes to @nickb1080

Changes proposed:
- use getBounding rect to determine if an element if visible or not

Upgrade Path (for changed or removed APIs):
- no breaking changes

Acceptance Checklist:
- [x] All commits have been squashed to one.
- [x] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [x] Documentation (README.md) and examples have been updated as needed.
- [x] If this is a code change, a spec testing the functionality has been added.
- [x] If the commit message has [changed] or [removed], there is an upgrade path above.

### CODE REVIEW

**Comment by nickb1080** [comment]
nice! thanks for putting this together, I totally dropped the ball

File: lib/components/ModalPortal.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ var ModalPortal = module.exports = React.createClass({
     return !this.props.isOpen && !this.state.beforeClose;
   },
 
-  contentHasFocus() {
+  contentHasFocus: function() {
```
Comment (conlanpatrek) [inline lib/components/ModalPortal.js]: 👍  Thank you. This was my mistake.

**Comment by sashashakun** [comment]
Hi all!

@claydiffrient, I use react-modal in my project and also have problem with this issue. Can you please give me some advice how to test this PR? I will be glad to write PR with testing, if this is the main reason not to merge it)

**Comment by claydiffrient** [comment]
@sashashakun Yeah testing would be great to have in place here.  It doesn't look like we have any specs that directly test the helpers.  So we'd probably want to create something like that such as `helpers.spec.js` or similar.  I guess technically they are more of a private detail so maybe the best practice would be to set up a modal in`Modal.spec.js` that might test it out fully.  

I personally think that the best approach would be to test it directly because the setup for everything would maybe outweigh the benefit of the test.

**Comment by conlanpatrek** [comment]
@claydiffrient - Looks like another PR was opened to resolve the missing tests #266

Is that enough to move forward with this?

**Comment by michaelckelly** [comment]
@evanhobbs @claydiffrient - Any chance of getting this to a merge-able state? Anything I can do to help get us there? This bug is impacting us and we'd love to get it fixed.

**Comment by everettwho** [comment]
@claydiffrient It seems like this is bug that's effecting a decent number of people using this module.  It looks like all the required items are ready. Is there anything else required that might be keeping you from merging this PR?

**Comment by diasbruno** [comment]
#521 and #271 were the new updates to tab engine.

I'm closing this for clean up. Please, feel free to reopen this if the problem persists. 
Thank you.

**Pull request closed without merging**